File: Two Pointers/1768. Merge Strings Alternately/Solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        // Time Complexity: O(max(n1, n2))
        // Space Complexity: O(n1 + n2)
        int n1 = word1.length();
        int n2 = word2.length();
        int l = 0;
        int r = 0;

        StringBuilder mergedString = new StringBuilder();
        while (l < n1 && r < n2){
            mergedString.append(word1.charAt(l));
            mergedString.append(word2.charAt(r));
            l++;
            r++;
        }

        while (l < n1){
            mergedString.append(word1.charAt(l));
            l++;
        }

        while (r < n2){
            mergedString.append(word2.charAt(r));
            r++;
        }
        return mergedString.toString();
    }
}

File: Two Pointers/1768. Merge Strings Alternately/Solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        # Time Complexity: O(max(n1, n2))
        # Space Complexity: O(n1 + n2)
        n1 = len(word1)
        n2 = len(word2)
        l = 0
        r = 0

        res = ""
        while l < n1 and r < n2:
            res += word1[l]
            res += word2[r]
            l += 1
            r += 1
        
        while l < n1:
            res += word1[l]
            l += 1
        
        while r < n2:
            res += word2[r]
            r += 1
        return res